.travis.yml

```diff
@@ -5,15 +5,47 @@ install:
   - curl http://download.opensuse.org/repositories/home:/laszlo_budai:/syslog-ng/xUbuntu_12.04/Release.key | sudo apt-key add -
   - echo "deb http://download.opensuse.org/repositories/home:/laszlo_budai:/syslog-ng/xUbuntu_12.04 ./" | sudo tee --append /etc/apt/sources.list.d/syslog-ng-obs.list
   - sudo apt-get update -qq
-  - sudo apt-get install -qq pkg-config flex bison xsltproc docbook-xsl libevtlog-dev libnet1-dev libglib2.0-dev libdbi0-dev libssl-dev libjson0-dev libwrap0-dev libpcre3-dev libcap-dev libesmtp-dev libgeoip-dev libhiredis-dev sqlite3 libdbd-sqlite3 libriemann-client-dev openjdk-7-jdk gradle-2.2.1 autoconf-archive
+  - sudo apt-get install -qq pkg-config flex bison xsltproc docbook-xsl libevtlog-dev libnet1-dev libglib2.0-dev libdbi0-dev libssl-dev libjson0-dev libwrap0-dev libpcre3-dev libcap-dev libesmtp-dev libgeoip-dev libhiredis-dev sqlite3 libdbd-sqlite3 libriemann-client-dev openjdk-7-jdk gradle-2.2.1 autoconf-archive # libsystemd-journal-dev libsystemd-daemon-dev
   - sudo pip install -r requirements.txt
 before_script:
   - ./autogen.sh
   - unset PYTHON_CFLAGS # HACK
-  - ./configure CFLAGS=-Werror --with-ivykis=internal --prefix=$HOME/install/syslog-ng --enable-pacct --enable-manpages --with-docbook=/usr/share/xml/docbook/stylesheet/docbook-xsl/manpages/docbook.xsl 
+  - ./configure
+      CFLAGS=-Werror
+      --with-ivykis=internal
+      --with-mongoc=internal
+      --with-librabbitmq=internal
+      --with-jsonc=internal
+      --prefix=$HOME/install/syslog-ng
+      --disable-env-wrapper
+      --disable-memtrace
+      --enable-ipv6
+      --enable-tcp-wrapper
+      --enable-spoof-source
+      --disable-sun-streams
+      --enable-sql
+      --enable-pacct
+      --enable-linux-caps
+      --enable-mongodb
+      --enable-json
+      --enable-amqp
+      --enable-stomp
+      --enable-smtp
+      --enable-redis
+      --disable-systemd
+      --enable-geoip
+      --enable-riemann
+      --enable-systemd-journal=optional
+      --enable-python
+      --enable-manpages
+      --enable-java
+      --enable-java-modules
+      --enable-native
+      --enable-all-modules
```

Comment:
We need to make explicit that we test all (possible) modules, and this also
documents which ones are untested. The script needs to consider this from a
black box perspective, and it must assume that unspecified modules will be
set to auto by the configure script.

On Fri, Apr 1, 2016 at 6:48 AM, Budai Laszlo <notifications@github.com>
wrote:

> In .travis.yml
> <https://github.com/balabit/syslog-ng/pull/982#discussion_r58162162>:
>
> > +      --enable-mongodb
> > +      --enable-json
> > +      --enable-amqp
> > +      --enable-stomp
> > +      --enable-smtp
> > +      --enable-redis
> > +      --disable-systemd
> > +      --enable-geoip
> > +      --enable-riemann
> > +      --enable-systemd-journal=optional
> > +      --enable-python
> > +      --enable-manpages
> > +      --enable-java
> > +      --enable-java-modules
> > +      --enable-native
> > +      --enable-all-modules
>
> Why do you need these explicit enables?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/balabit/syslog-ng/pull/982/files/b1430a34c584a03477fc4b2c7b64f86fbb1452c3#r58162162>
>